src/wasm.h

```diff
@@ -1137,61 +1137,62 @@ class AllocatingModule : public Module {
 // etc.
 //
 
-template<class ReturnType>
+template<typename SubType, typename ReturnType>
 struct WasmVisitor {
   virtual ~WasmVisitor() {}
   // should be pure virtual, but https://gcc.gnu.org/bugzilla/show_bug.cgi?id=51048
   // Expression visitors
-  virtual ReturnType visitBlock(Block *curr) { abort(); }
-  virtual ReturnType visitIf(If *curr) { abort(); }
-  virtual ReturnType visitLoop(Loop *curr) { abort(); }
-  virtual ReturnType visitBreak(Break *curr) { abort(); }
-  virtual ReturnType visitSwitch(Switch *curr) { abort(); }
-  virtual ReturnType visitCall(Call *curr) { abort(); }
-  virtual ReturnType visitCallImport(CallImport *curr) { abort(); }
-  virtual ReturnType visitCallIndirect(CallIndirect *curr) { abort(); }
-  virtual ReturnType visitGetLocal(GetLocal *curr) { abort(); }
-  virtual ReturnType visitSetLocal(SetLocal *curr) { abort(); }
-  virtual ReturnType visitLoad(Load *curr) { abort(); }
-  virtual ReturnType visitStore(Store *curr) { abort(); }
-  virtual ReturnType visitConst(Const *curr) { abort(); }
-  virtual ReturnType visitUnary(Unary *curr) { abort(); }
-  virtual ReturnType visitBinary(Binary *curr) { abort(); }
-  virtual ReturnType visitSelect(Select *curr) { abort(); }
-  virtual ReturnType visitHost(Host *curr) { abort(); }
-  virtual ReturnType visitNop(Nop *curr) { abort(); }
-  virtual ReturnType visitUnreachable(Unreachable *curr) { abort(); }
+  ReturnType visitBlock(Block *curr) { abort(); }
```

Comment:
Do you want to default to `abort()`? Or do-nothing?

It could be a template parameter as well if both make sense:
```
enum class VisitorDefaultBehavior { Abort, NoOp };
template<typename SubType, typename ReturnType = void, VisitorDefaultBehavior Behavior = VisitorDefaultBehavior::Abort>
struct WasmVisitor {
  // ...
private:
  void Default() {
    switch (Behavior) {
    case VisitorDefaultBehavior::Abort: abort();
    case VisitorDefaultBehavior::NoOp: break;
    }
  }
  ReturnType visit*(...) {  Default(); }
  // ...
};
```